### PR TITLE
Add support for apt/yum repos & Shield parameters

### PR DESCRIPTION
**Pull request by pabrahamsson**
This adds support for using the official apt/yum repos (copied from [elasticsearch/puppet-logstash](https://github.com/elasticsearch/puppet-logstash)).
It also exposes a few additional parameters necessary for Shield.

### CODE REVIEW

**Comment by jlambert121** [comment]
Could you add tests around the new parameters and fix the broken ones?

**Comment by pabrahamsson** [comment]
B0rked tests fixed and additional test coverage added.

File: manifests/init.pp [hunk 1]
```diff
@@ -47,9 +78,48 @@
...
+          $_provider     = 'apt'
+        }
+        'RedHat': {
+          $_package_name = 'python-elasticsearch-curator'
+          $_provider     = 'yum'
+        }
+        default: {
+          $_package_name = 'elasticsearch-curator'
+          $_provider     = 'pip'
+          #$bin_file     = '/bin/curator'
```
Comment (jlambert121) [inline manifests/init.pp]: Leftover line to clean up?

File: manifests/init.pp [hunk 2]
```diff
@@ -47,9 +78,48 @@
     }
   }
 
-  package { $package_name:
-    ensure   => $ensure,
-    provider => $provider,
+  case $manage_repo {
```
Comment (jlambert121) [inline manifests/init.pp]: Could you write some tests around manage_repo?

**Comment by jlambert121** [comment]
Thanks for this, looks like a great addition!  Could you add a couple more tests to this and we'll be good to go?

File: manifests/init.pp (hunk 2 above)
Comment (pabrahamsson) [inline manifests/init.pp]: I had added some in `spec/classes/curator_repo_spec.rb`. Did you have more in mind?

**Comment by jlambert121** [comment]
I think that's good, thanks.  It looks like it needs a rebase though as there is some overlap with #63, sorry about that.

**Comment by pabrahamsson** [comment]
oh boy, this got messy. I'll need to squash this one before you merge but am having some issues. I'll let you know when it's ready to be merged.

**Comment by pabrahamsson** [comment]
Alright @jlambert121 I think we're good to go.

**Comment by jlambert121** [comment]
It did get messy, sorry about that - I forgot there were 2 PRs that would overlap.  Thansk for reworking it!

**Pull request merged**